src/components/CGroupTable.vue

```diff
@@ -0,0 +1,119 @@
+<template>
+  <div class="px-4 pt-2">
+    <table class="w-full bg-white rounded mb-4">
+      <thead>
+        <tr class="bg-gray-100 h-8">
+          <c-th
+            v-for="(column, index) in columns"
+            :label="column.label"
+            :name="column.name"
+            :key="index"
+          ></c-th>
+        </tr>
+      </thead>
+      <tbody>
+        <template v-for="g in groups">
+          <tr :key="g.groupId" @click="toggleRow(g.groupId)" class="text-xs hover:bg-blue-100">
+            <td class="border px-4 py-2">
+              <div class="flex">
+                <img
+                  v-if="openedRows.includes(g.groupId)"
+                  src="@/assets/img/expand_more.svg"
+                  width="20rem"
+                />
+                <img v-else src="@/assets/img/expand_less.svg" width="20rem" />
+                <pre class="text-xs ml-2 whitespace-normal">{{ g.groupId }}</pre>
+              </div>
+            </td>
+            <td class="border px-4 py-2">
+              {{ remainingTime(g.messages) }}
+            </td>
+            <td class="border px-4 py-2">
+              {{ progress(g.messages) | percentage }}
+            </td>
+          </tr>
+          <tr :key="g.groupId + 0" v-if="openedRows.includes(g.groupId)">
+            <td class="border text-xs px-4 py-2" :colspan="columns.length">
+              <c-group-row
+                :key="g.groupId"
+                :messages="g.messages"
+                :groupId="g.groupId"
+              ></c-group-row>
+            </td>
+          </tr>
+        </template>
+      </tbody>
+    </table>
+  </div>
+</template>
+<script>
+import CGroupRow from '@/components/CGroupRow';
+import CTh from '@/components/CTh';
+import { mapState } from 'vuex';
+import utils from '@/utils';
+
+export default {
+  name: 'CGroupTable',
+  components: { CGroupRow, CTh },
+  data() {
+    return {
+      columns: [
+        { label: 'Group Id', name: 'groupId' },
+        { label: 'Remaining Time', name: 'remainingTime' },
+        { label: 'Progress', name: 'progress' }
+      ],
+      openedRows: []
+    };
+  },
+  methods: {
+    toggleRow(id) {
+      const index = this.openedRows.indexOf(id);
+      if (index >= 0) {
+        this.openedRows = this.openedRows.filter(item => item !== id);
+      } else {
+        this.openedRows = [...this.openedRows, id];
+      }
+    },
+    remainingTime(messages) {
+      // Get the largest remaining time (suppossing they are running in parallel)
+      // TODO: add heuristic to add a value based on pending messages
```

Comment:
enquedDatetime of a group should be almost the same so 
then
```
const progress = progress(messages)
const factor = (1 - progress) / progress;
const diff = utils.formatMillis((this.currentDate - enquedDatetime) * factor)
```
should be precise enougth